### PR TITLE
Move test helpers into common modules

### LINKED ISSUE #1031: test helpers
As discussed previously with @kevinbarabash, we should move the helper functions in `katex-spec.js` like `toParse`, `toNotParse`, and `toParseLike` into a common module shared by all test modules.

I'd also like to suggest a `toBuildLike` which compares the HTML outputs.  I think we do this in various places by manually calling `getBuilt` twice and comparing.

### PR DESCRIPTION
**Pull request by edemaine**
Fix #1031 
* `helpers.js` gets all the helper functions
* `setup.js` gets the common Jest setup (serializer, expect extensions)

This PR is basically just refactoring; I did very little to clean up the code itself.  But the hope is that, by putting all the shared code (e.g. `expect` extensions) into a common module, it'll be easier to do test code cleanup.

### CODE REVIEW

**Comment by edemaine** [comment]
@kevinbarabash Alternatively, we could add tests to test the test framework. 😄 
Shall I change that in this PR?

**Comment by kevinbarabash** [comment]
> Shall I change that in this PR?

May as well.  Thanks. :)

File: test/helpers.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    try {
+        return _getBuilt(expr, settings || defaultSettings);
+    } catch (e) {
+        result.pass = false;
+        if (e instanceof ParseError) {
+            result.message = () => "'" + expr + "' failed " +
+                "parsing with error: " + e.message;
+        } else {
+            result.message = () => "'" + expr + "' failed " +
+                "parsing with unknown error: " + e.message;
```
Comment (ylemkimon) [inline test/helpers.js]: `parsing` -> `building` (same for above)

File: test/setup.js [hunk 2]
```diff
@@ -2,18 +2,239 @@
...
+        } catch (e) {
+            result.pass = false;
+            if (e instanceof ParseError) {
+                result.message = () => "'" + actual + "' failed to " +
+                    "build with error: " + e.message;
+            } else {
+                result.message = () => "'" + actual + "' failed " +
+                    "building with unknown error: " + e.message;
+            }
+        }
```
Comment (ylemkimon) [inline test/setup.js]: Why not use `buildAndSetResult`?

File: test/helpers.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import parseTree from "../src/parseTree";
+import Settings from "../src/Settings";
+
+export const defaultSettings = new Settings({
+    strict: false, // deal with warnings only when desired
+});
+export const strictSettings = new Settings({strict: true});
+
+export const _getBuilt = function(expr, settings) {
+    const usedSettings = settings ? settings : defaultSettings;
```
Comment (ylemkimon) [inline test/helpers.js]: I think we can use `||` and overwrite parameter, i.e., `settings = settings || defaultSetting`.

**Comment by kevinbarabash** [comment]
I think it's probably best to DRY the code up in a separate PR.  I find reviewing changes were stuff is moved and refactored at the same time hard to follow.

File: package.json [hunk 4]
```diff
@@ -87,6 +87,10 @@
     "nomnom": "^1.8.1"
   },
   "jest": {
+    "collectCoverageFrom": [
+      "src/**/*.js",
+      "contrib/**/*.js"
```
Comment (kevinbarabash) [inline package.json]: Thanks!

**Comment by kevinbarabash** [comment]
@edemaine there's some lint and a couple of failing tests.

**Comment by ylemkimon** [comment]
@edemaine I'll do it in another PR 😄
